lib/ansible/plugins/connection/ssh.py

```diff
@@ -459,15 +459,26 @@ def _handle_error(
         # No exception is raised, so the connection is retried - except when attempting to use
         # sshpass_prompt with an sshpass that won't let us pass -P, in which case we fail loudly.
         elif return_tuple[0] in [1, 2, 3, 4, 6]:
-            msg = 'sshpass error:'
+            error_code = return_tuple[0]
+            cmd_error = to_native(return_tuple[2]).rstrip()
+            error_messages = {
+                1: "sshpass: Invalid command line argument. Command given ({})".format(cmd_error),
```

Comment:
I would avoid embedding the stderr for this and the next one in case it contains sensitive information that should not be exposed in `no_log`.